pkg/apis/extensions/validation/validation.go

```diff
@@ -134,13 +134,18 @@ func validateDaemonSetStatus(status *extensions.DaemonSetStatus, fldPath *field.
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(status.UpdatedNumberScheduled), fldPath.Child("updatedNumberScheduled"))...)
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(status.NumberAvailable), fldPath.Child("numberAvailable"))...)
 	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(status.NumberUnavailable), fldPath.Child("numberUnavailable"))...)
+	allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(status.NumberUnavailable), fldPath.Child("numberUnavailable"))...)
+	if status.Uniquifier != nil {
+		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(*status.Uniquifier), fldPath.Child("uniquifier"))...)
+	}
 	return allErrs
 }
 
 // ValidateDaemonSetStatus validates tests if required fields in the DaemonSet Status section
 func ValidateDaemonSetStatusUpdate(ds, oldDS *extensions.DaemonSet) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&ds.ObjectMeta, &oldDS.ObjectMeta, field.NewPath("metadata"))
 	allErrs = append(allErrs, validateDaemonSetStatus(&ds.Status, field.NewPath("status"))...)
+	// TODO: validate that status.uniquifier is never decremented
```

Comment:
This makes sure that controller doesn't set it to a smaller number because of stale cache.